packages/@aws-cdk/aws-rds/lib/props.ts

```diff
@@ -1,15 +1,31 @@
 import ec2 = require('@aws-cdk/aws-ec2');
 import kms = require('@aws-cdk/aws-kms');
 import { SecretValue } from '@aws-cdk/cdk';
+import { SecretRotationApplication } from './secret-rotation';
 
 /**
- * The engine for the database cluster
+ * A database cluster engine. Provides mapping to the serverless application
+ * used for secret rotation.
  */
-export enum DatabaseClusterEngine {
-  Aurora = 'aurora',
-  AuroraMysql = 'aurora-mysql',
-  AuroraPostgresql = 'aurora-postgresql',
-  Neptune = 'neptune'
+export class DatabaseClusterEngine {
+  public static readonly Aurora = new DatabaseClusterEngine('aurora', SecretRotationApplication.MysqlRotationSingleUser);
+  public static readonly AuroraMysql = new DatabaseClusterEngine('aurora-mysql', SecretRotationApplication.MysqlRotationSingleUser);
+  public static readonly AuroraPostgresql = new DatabaseClusterEngine('aurora-postgresql', SecretRotationApplication.PostgresRotationSingleUser);
+
+  /**
+   * The engine.
+   */
+  public readonly engine: string;
```

Comment:
The class name tells me the instance of the class *is* an engine... So this is... the engine's engine? That reads wrong. Maybe call this `name`?

Reply:
Thanks, I was obviously lacking imagination here 😄